demo/lightning/checkpoint/multinode/train.py

```diff
@@ -128,7 +128,22 @@ def __init__(
     ) -> None:
         super().__init__()
         self.optimizer = optimizer
-        self.model = Transformer(vocab_size=vocab_size, nlayers=nlayers)
+        self.model = None
+        self.vocab_size = vocab_size
+        self.nlayers = nlayers
+
+    # Initialize the model here to allow it to be initialized on the GPU; see
+    # https://lightning.ai/docs/pytorch/stable/advanced/model_parallel/fsdp.html#speed-up-model-initialization
+    def configure_model(self):
+        if self.model is not None:
+            return
+        # Use the nhid, ninp, and nhead parameters from the guide linked above.
+        # This means that 1B parameters corresponds to 32 layers.
+        self.model = Transformer(vocab_size=self.vocab_size,
+                                 nlayers=self.nlayers,
```

Comment:
any reason not to make similar changes in single node ?

Reply:
Sure, I will follow up on that in another PR